### PR TITLE
Close open PR if Discourse and Github content match

### PR DESCRIPTION
**Pull request by deusebio**
I have noticed that a good feature could be to close open PR if github and Discourse matches. This should provide an automatic way to resolve conflicts or also revert PR opened that, given new tags, are no longer necessary

### CODE REVIEW

File: src/__init__.py [hunk 1]
```diff
@@ -139,15 +141,18 @@ def run_migrate(clients: Clients, user_inputs: UserInputs) -> dict[str, str]:
             user_inputs.commit_sha,
             DOCUMENTATION_TAG,
         )
+        # If a PR is open we should probably close the PR
```
Comment (jdkandersson) [inline src/__init__.py:144]: Perhaps `Given there are diffs compared to the base, if a PR is open, it should be closed`?

File: tests/unit/test___init__.py [hunk 2]
```diff
@@ -731,3 +734,59 @@ def test_run_migrate_same_content_local_and_server(caplog, mocked_clients):
 
     assert not returned_migration_reports
     assert any("No community contribution found" in record.message for record in caplog.records)
+    assert not mock_edit_pull_request.called
+
+
+@mock.patch("src.repository.Client.get_pull_request")
```
Comment (jdkandersson) [inline tests/unit/test___init__.py:740]: Should we use monkeypatch: https://docs.pytest.org/en/7.1.x/how-to/monkeypatch.html

File: tests/unit/test___init__.py [hunk 3]
```diff
@@ -731,3 +734,59 @@ def test_run_migrate_same_content_local_and_server(caplog, mocked_clients):
...
+    caplog.clear()
+    with caplog.at_level(logging.INFO):
+        # run is repeated in unit tests / integration tests
+        returned_migration_reports = run_migrate(
+            clients=mocked_clients, user_inputs=user_inputs
+        )  # pylint: disable=duplicate-code
+
+    assert not returned_migration_reports
+    assert any("No community contribution found" in record.message for record in caplog.records)
+    assert mock_edit_pull_request.called
```
Comment (gregory-schiano) [inline tests/unit/test___init__.py:792]: Shouldn't we also test that the method was called with the state attribute equal to "closed" ?

File: src/__init__.py (hunk 1 above)
Comment (deusebio) [reply src/__init__.py:144]: I'd say

```Given there are NO diffs compared to the base, if a PR is open, it should be closed```

File: tests/unit/test___init__.py (hunk 2 above)
Comment (deusebio) [reply tests/unit/test___init__.py:740]: Uhm, I kinda used the one that was used elsewhere in the file to be consistent. It seems that we are using - depending on the file - either monkeypatch or mock.patch, not sure why. But I would keep it consistent per file to start with, but then we should pick one of the two and refactor the others across.

File: tests/unit/test___init__.py (hunk 3 above)
Comment (deusebio) [reply tests/unit/test___init__.py:792]: yes, fine.

File: tests/unit/test___init__.py (hunk 2 above)
Comment (jdkandersson) [reply tests/unit/test___init__.py:743]: yeh, monkeypatch is preferred. No idea where mock.patch came from

**Review by jdkandersson — DISMISSED** [review]

**Review by jdkandersson — DISMISSED** [review]

**Review by jdkandersson — APPROVED** [review]

**Pull request merged**